components/esm/loadscpt.cpp

```diff
@@ -30,10 +30,11 @@ namespace ESM
         // The tmp buffer is a null-byte separated string list, we
         // just have to pick out one string at a time.
         char* str = &tmp[0];
-        if (!str) {
+        if (!str && mVarNames.size()) > 0
```

Comment:
A typo! Last bracket should be after 0.
Also please squash your commits to one - otherwise here will be six commit to change a couple of lines.